### PR TITLE
Added 'last_value' to aggregate function list

### PR DESCRIPTION
**Pull request by Gutzeit555**
fixes Issue 6623 in the qgis/QGIS-Documentation repository
I made further changes to the list, as it did not seem up to date to me:
added „first_value“
deleted „stdevsample“
renamed „min“ to „minimum“
renamed „max“ to „maximum“

## Description

[Replace this with some text explaining the rationale and details about this pull request]

### CODE REVIEW

**Comment by m-kuhn** [comment]
https://github.com/qgis/QGIS-Documentation/issues/6623
CC  @nirvn

**Review by nirvn — APPROVED** [review]
Nice papercut fix.

**Comment by DelazJ** [comment]
Unless I miss recent changes, are you sure of these? It doesn't look enough to me. Using max/min/stdevsample in an expression dialog, I can get output while maximum/minimum return invalid expression in the dialog. Same for first_value and last_value. No such aggregates.

@nirvn, any reason there's no direct first_value and last_value functions in the aggregates group?

**Comment by Gutzeit555** [comment]
@DelazJ 
This is the list from the drop-down menu for "Aggregate Function" in the dialog I get from „Aggregate“ in the Processing Toolbox (see picture). I tried the functions described and they worked. I assumed that this list is complete and correct and that the same function is also used in the expression dialog, also because the link in section "25.1.16.3. Aggregate" leads there. Is this not correct?
aggregate

**Comment by nyalldawson** [comment]
> Unless I miss recent changes, are you sure of these? It doesn't look enough to me. Using max/min/stdevsample in an expression dialog, I can get output while maximum/minimum return invalid expression in the dialog. Same for first_value and last_value. No such aggregates.

The change IS incorrect -- it's 100% "min" and "max" (direct from the code, and see eg https://github.com/qgis/QGIS/blob/master/tests/src/core/testqgsexpression.cpp#L2349)

File: resources/function_help/json/aggregate [hunk 1]
```diff
@@ -5,7 +5,7 @@
...
+	{"arg":"aggregate", "description":"a string corresponding to the aggregate to calculate. Valid options are:<br /><ul><li>count</li><li>count_distinct</li><li>count_missing</li><li>minimum</li><li>maximum</li><li>sum</li><li>mean</li><li>median</li><li>stdev</li><li>range</li><li>minority</li><li>majority</li><li>q1: first quartile</li><li>q3: third quartile</li><li>iqr: inter quartile range</li><li>min_length: minimum string length</li><li>max_length: maximum string length</li><li>concatenate: join strings with a concatenator</li><li>concatenate_unique: join unique strings with a concatenator</li><li>collect: create an aggregated multipart geometry</li><li>array_agg: create an array of aggregated values</li><li>first_value</li><li>last_value</li></ul>"},
```
Comment (nyalldawson) [inline resources/function_help/json/aggregate:8]:
```suggestion
	{"arg":"aggregate", "description":"a string corresponding to the aggregate to calculate. Valid options are:<br /><ul><li>count</li><li>count_distinct</li><li>count_missing</li><li>min</li><li>max</li><li>sum</li><li>mean</li><li>median</li><li>stdev</li><li>range</li><li>minority</li><li>majority</li><li>q1: first quartile</li><li>q3: third quartile</li><li>iqr: inter quartile range</li><li>min_length: minimum string length</li><li>max_length: maximum string length</li><li>concatenate: join strings with a concatenator</li><li>concatenate_unique: join unique strings with a concatenator</li><li>collect: create an aggregated multipart geometry</li><li>array_agg: create an array of aggregated values</li><li>first_value</li><li>last_value</li></ul>"},
```

**Review by nyalldawson — CHANGES_REQUESTED** [review]

**Comment by nyalldawson** [comment]
This PR is incorrect -- none of the changes are valid

**Comment by Gutzeit555** [comment]
I'm not sure I understand: are there different functions used in the Processing Toolbox and the expression editor?
btw: I don't know if it's important, but in the [aggregates group](https://docs.qgis.org/testing/en/docs/user_manual/expressions/functions_list.html#id74) it also says minimum and maximum.

**Comment by DelazJ** [comment]
@Gutzeit555 The link in the [aggregate algorithm docs](https://docs.qgis.org/3.22/en/docs/user_manual/processing_algs/qgis/vectorgeometry.html#aggregate) points to the [aggregates group](https://docs.qgis.org/3.22/en/docs/user_manual/expressions/functions_list.html#aggregates-functions) (which the drop-down list in the alg derives from) and not the [aggregate function](https://docs.qgis.org/3.22/en/docs/user_manual/expressions/functions_list.html#aggregate) that this PR is modifying.
The issue with qgis/QGIS-Documentation#6623 is that there's no "first value" and "last value" function in the aggregates group functions. They exist nowhere but in that algorithm, and we don't want to document (if needed) them there. Reason why I was asking whether there's a reason they do not exist as expression function (_replying to myself, maybe because of array_first and array_last?_) .

**Comment by DelazJ** [comment]
The other issue I see your PR raise is why the aggregate function doesn't have minimum and maximum as aggregate parameter (all the other parameters name matches an existing aggregate function) instead of min and max.

**Comment by Gutzeit555** [comment]
Ok, I understood my mistake, everything is wrong. I'm really sorry about the confusion!
What can I do to cancel this PR?

**Comment by m-kuhn** [comment]
Let's close this. A new issue has been opened for aliases.
@Gutzeit555 If you have more ideas to for fixes, please open a PR again any time.

**Comment by DelazJ** [comment]
> Ok, I understood my mistake, everything is wrong. I'm really sorry about the confusion!

No worries. I can't count the number of "wrong" PRs I've opened :smile:. The mismatch of parameters is confusing and you help find an inconsistency in the function, anyway.
And as @m-kuhn said, do not hesitate to open new PR if you want to fix anything else.

**Comment by Gutzeit555** [comment]
Thank you!  I will keep trying ;-)

**Pull request closed without merging**